packages/fuels-contract/src/script_calls.rs

```diff
@@ -0,0 +1,181 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
+
+use fuel_gql_client::fuel_tx::{Output, Receipt, Transaction};
+use fuel_tx::Input;
+
+use crate::contract::get_decoded_output;
+
+use fuels_core::{
+    parameters::{CallParameters, TxParameters},
+    Tokenizable,
+};
+use fuels_signers::{provider::Provider, WalletUnlocked};
+
+use fuels_types::{errors::Error, param_types::ParamType};
+
+use crate::execution_script::{CompiledScript, TransactionExecution};
+
+pub struct ScriptInterface {
+    pub script: TransactionExecution,
+    pub wallet: WalletUnlocked,
+}
+
+#[derive(Debug)]
+pub struct ScriptCallResponse<D> {
+    pub value: D,
+    pub receipts: Vec<Receipt>,
+    pub gas_used: u64,
+}
+
+impl<D> ScriptCallResponse<D> {
+    /// Get the gas used from ScriptResult receipt
+    fn get_gas_used(receipts: &[Receipt]) -> u64 {
+        receipts
+            .iter()
+            .rfind(|r| matches!(r, Receipt::ScriptResult { .. }))
+            .expect("could not retrieve ScriptResult")
+            .gas_used()
+            .expect("could not retrieve gas used from ScriptResult")
+    }
+
+    pub fn new(value: D, receipts: Vec<Receipt>) -> Self {
+        Self {
+            value,
+            gas_used: Self::get_gas_used(&receipts),
+            receipts,
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Contains all data relevant to a single script call
+pub struct ScriptCall {
```

Comment:
In the case of contracts, we keep `ContractCall` and `ContractCallHandler` separate to make multi-calls easier to construct. In the case of scripts, we don't have that requirement, so we could combine them into a single struct. 

Wether that is a cleaner design, in the end, is for you to decide, but we should hide `ScriptCall` from the user at any rate to avoid confusion.

Ideally we then have `ScriptCallHandler` (or maybe `ScriptHandler`) as the only entry point to doing something with scripts in the SDK. It replaces the functionality of `run_compiled_script` and enables main arguments.

Reply:
It is possible. But if you have a script that does multi-calls, all the relevant information is contained by a single `ScriptCall`:
- script binary
- script data
- inputs
- outputs

`ContractCall` holds information specific to a single call e.g. selector, contract id, output variables. Note all of these would be contained in the above fields if we use 'ScriptCall'.

Reply:
Got it, thanks for clarifying! I'm removing it, I had added it for extensibility.